app/components/provider_interface/activity_log_event_component.rb

```diff
@@ -58,17 +58,20 @@ def event_description
     def course_option
       current_status = event.application_status_at_event || application_choice.status
 
-      @course_option ||= if changes['offer_changed_at'].present? && changes['offered_course_option_id'].present?
-                           from_event = CourseOption.find_by(id: changes['offered_course_option_id'].second)
-                           from_event || application_choice.offered_option
+      changed_option_id = changes['current_option_id']&.second
+      changed_option_id ||= changes['offered_course_option_id']&.second
+      option_at_time = CourseOption.find_by(id: changed_option_id)
+
+      @course_option ||= if changes['offer_changed_at'].present? && option_at_time.present?
+                           option_at_time
                          elsif current_status == 'offer'
-                           # older offers won't have a course option id as part of this
-                           from_event = CourseOption.find_by(id: changes['offered_course_option_id']&.second)
-                           from_event || application_choice.offered_option
+                           # older offers may not have option_at_time available
+                           # FIXME: should not be possible after data migration
```

Comment:
Could you explain the logic here a bit more?

If the audit log contains changes to either the `current_option_id` or `offered_course_option_id` fields, then `option_at_time` contains whichever course we changed to. If neither of those are present (ie because the choice is from before we started always setting `offered_course_option_id`) then the offer hasn't changed, and we simply show whatever option the `application_choice` has now along with the `offer` event? I'm missing something here — what's not possible after the data migration?